### PR TITLE
Referrals updates 

### PR DESCRIPTION
**Pull request by dylanlott**
Referrals updates

### CODE REVIEW

**Review by bryanchriswhite — APPROVED** [review]
I'd like to understand a little bit more about how the referrals model is used/works but otherwise these seem to be reasonable changes

**Comment by phutchins** [comment]
LGTM. Merging.

**Comment by braydonf** [comment]
Is this not work-in-progress (e.g. not ready to merge) as with the WIP in the title? Looks like there is API/model change here that will need major bump, and code coverage decreased.

**Comment by phutchins** [comment]
@dylanlott & @barbaraliau , can you please remove WIP from this MR and discuss with @braydonf the API change and where the version should land for this? I can re-publish (unpublish and publish) the version that is decided upon.

File: lib/models/referral.js [hunk 1]
```diff
@@ -139,20 +145,34 @@ Referral.statics.create = function(marketing, recipientEmail, type) {
...
-        email: recipientEmail
-      },
-      type
-    });
+    Referral.findOne({
+      'sender.id': marketing._id,
+      'recipient.email': recipientEmail
+    }).then((referral) => {
+      if (referral) {
+        referral.count += referral.count;
```
Comment (braydonf) [inline lib/models/referral.js:153]: There may be a concurrency issue here if two processes do this at the same time, one of the count will go missing.

**Pull request merged**